### PR TITLE
Fix viewers getting logged out when changing text temp var

### LINKED ISSUE #3913: using the text input variable as a viewer logs the user out
to repro, log into chronograf as a viewer, then open a dashboard with a text input variable. click the text box of the template variable and notice that you are logged out.

fyi @chnn

### PR DESCRIPTION
**Pull request by ischolten**
Closes #3913

_What was the problem?_
Updating the text in the text template variable updated the db which caused an error if the user was a visitor.
_What was the solution?_
Updating the text in the text template variable updates the locally selected value and the url param but does not overwrite the value in the db.

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/dashboards/reducers/ui.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import {NULL_HOVER_TIME} from 'src/shared/constants/tableGraph'
 import {DashboardUIState} from 'src/types/dashboards'
 import {Action, ActionType} from 'src/dashboards/actions'
 
+import {TemplateType} from 'src/types/tempVars'
```
Comment (alexpaxton) [inline ui/src/dashboards/reducers/ui.ts]: Much type very strong

File: ui/src/dashboards/utils/tempVars.ts [hunk 2]
```diff
@@ -41,19 +41,27 @@ export const applyLocalSelections = (
...
+  for (const {tempVar, values, type} of templates) {
     if (!values.length) {
       continue
     }
 
     // Attempt to use supplied selection
     let selection = selections[tempVar]
 
-    if (!selection || !values.find(v => v.value === selection)) {
+    if (
```
Comment (alexpaxton) [inline ui/src/dashboards/utils/tempVars.ts]: Death to ternaries

File: ui/test/tempVars/components/TextTemplateSelector.test.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import {TemplateType, TemplateValueType} from 'src/types'
 
 describe('TextTemplateSelector', () => {
   test('updates template on enter', () => {
-    const onUpdateTemplateMock = jest.fn()
+    const onPickValue = jest.fn()
```
Comment (alexpaxton) [inline ui/test/tempVars/components/TextTemplateSelector.test.tsx]: Should this have 'Mock' on the end?

Comment (ischolten) [reply ui/test/tempVars/components/TextTemplateSelector.test.tsx]: yea i can add it

**Pull request merged**